### ISSUE TITLE
RFE: stop silently ignoring `--now` option when combined with `systemctl preset`

### ISSUE THREAD

**Issue by sourcejedi** [systemd/systemd#8312]
### Submission type

 - Request for enhancement (RFE)

### systemd version the issue has been seen with

237-something from git

### Used distribution

Fedora

### RFE

`systemctl preset --now foo.service` silently ignores the `--now` option.  This seems clearly incorrect.

It should simply support the meaning of `--now`, to start or stop the service when the preset enables or disables a service.  Don't care too much what happens when the service is already enabled/disabled, but `disable --now` stops the unit even if it was already disabled, so it would be nice to be consistent with that.

Alternatively, it could warn that this feature will not be implemented.

**Comment by SjonHortensius** [comment]
I agree, I distribute site-packages with presets and I need to manually check all enabled units (to start them) or reboot the machine. `systemctl preset --now` would fix this

**Comment by belm0** [comment]
> disable --now stops the unit even if it was already disabled

This is not the behavior I observe, and actually was just about to file a bug about it.  If service status is disable but running, disable --now will not stop the service.

**Comment by sourcejedi** [comment]
@belm0 I retested on my machine and both enable and disable worked how I said.  Both on Fedora's v234, and a late v237-git. Feel free to open an issue with your steps to reproduce etc.  If you want you can @ me on it and I can add mine for comparison - because I'm not 100% clear on which behaviour would count as "correct", but it sounds very suspicious if it works completely opposite on different systems.

**Comment by YHNdnzj** [comment]
Fixed by f9c76997db9f13bcae020f899adb81e17b926dc5
